File: client/ios/Runner/Application/FlavorChannel.swift
import Foundation

public class FlavorChannel {
    public static func register(with messanger: FlutterBinaryMessenger) {
        let channel = FlutterMethodChannel(
            name: "com.gutlogic.app/flavor",
            binaryMessenger: messanger
        )
        channel.setMethodCallHandler(FlavorChannel().handle)
    }
    
    func handle(call: FlutterMethodCall, result: @escaping FlutterResult) {
        if (call.method == "get") {
            result(Bundle.main.infoDictionary?["Flavor"])
        } else {
            result(FlutterMethodNotImplemented)
        }
    }
}

File: client/android/app/src/main/kotlin/com/gutlogic/app/FlavorChannel.kt
package com.gutlogic.app

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class FlavorChannel(messenger: BinaryMessenger) {
    val channel = MethodChannel(messenger, "com.gutlogic.app/flavor")

    init {
        channel.setMethodCallHandler(::handle)
    }

    private fun handle(call: MethodCall, result: MethodChannel.Result) {
        if (call.method == "get") {
            result.success(BuildConfig.FLAVOR)
        } else {
            result.notImplemented()
        }
    }
}
